Extension/test/scenarios/SingleRootProject/tests/lmTool.test.ts

```diff
@@ -0,0 +1,366 @@
+    });
+
+    const testGetProjectContext = async ({
+        compiler,
+        expectedCompiler,
+        context,
+        compilerArguments: compilerArguments,
+        expectedCompilerArguments
+    }: {
+        compiler: string;
+        expectedCompiler: string;
+        context: { flags: Record<string, unknown> };
+        compilerArguments: string[];
+        expectedCompilerArguments: string[];
+    }) => {
+        arrangeProjectContextFromCppTools({
+            projectContextFromCppTools: {
+                language: 'cpp',
+                standardVersion: 'c++20',
+                compiler: compiler,
+                targetPlatform: 'windows',
+                targetArchitecture: 'x64',
+                fileContext: {
+                    compilerArguments: compilerArguments
+                }
+            }
+        });
+
+        const result = await getProjectContext(mockTextDocumentStub.uri, context, new vscode.CancellationTokenSource().token);
+
+        ok(telemetryStub.calledOnce, 'Telemetry should be called once');
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "language": 'C++'
+        })));
+        if (expectedCompiler) {
+            ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+                "compiler": expectedCompiler
+            })));
+        }
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "standardVersion": 'C++20'
+        })));
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "targetPlatform": 'Windows'
+        })));
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "targetArchitecture": 'x64'
+        })));
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "compilerArgumentCount": compilerArguments.length.toString()
+        })));
+        if (expectedCompilerArguments.length > 0) {
+            ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+                "filteredCompilerArguments": expectedCompilerArguments.join(' ')
+            })));
+        }
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            'time': sinon.match.string
+        })));
+        ok(result, 'result should not be undefined');
+        ok(result.language === 'C++');
+        ok(result.compiler === expectedCompiler);
+        ok(result.standardVersion === 'C++20');
+        ok(result.targetPlatform === 'Windows');
+        ok(result.targetArchitecture === 'x64');
+        ok(JSON.stringify(result.compilerArguments) === JSON.stringify(expectedCompilerArguments));
+    };
+
+    it('should log telemetry and provide cpp context properly when experimental flags are not defined.', async () => {
```

Comment:
Is this comment correct/intended? I don't see it logging anything. Should it be reworded to make that clearer?

Reply:
Updated, thanks.